### ISSUE TITLE
Building the sysimg on power fails in debug mode.

### ISSUE THREAD

**Issue by vchuravy** [JuliaLang/julia#18551]
I was looking at test failures that are happening on master on power and making a debug build failed for me.

This is on 628d0e4bd
```julia
markdown/Markdown.jl
docs/Docs.jl
/home/valentin/julia/base/precompile.jl
Return operand #2 has unhandled type i32
UNREACHABLE executed at /home/valentin/julia/deps/srccache/llvm-3.8.1/lib/CodeGen/CallingConvLower.cpp:114!

signal (6): Aborted
while loading no file, in expression starting on line 0
gsignal at /lib64/power8/libc.so.6 (unknown line)
abort at /lib64/power8/libc.so.6 (unknown line)
_ZN4llvm25llvm_unreachable_internalEPKcS1_j at /home/valentin/julia/usr/bin/../lib/libLLVM-3.8.so (unknown line)
_ZN4llvm7CCState13AnalyzeReturnERKNS_15SmallVectorImplINS_3ISD9OutputArgEEEPFbjNS_3MVTES7_NS_11CCValAssign7LocInfoENS2_10ArgFlagsTyERS0_E at /home/valentin/julia/usr/bin/../lib/libLLVM-3.8.so (unknown line)
unknown function (ip: 0x3fffa70106d3)
unknown function (ip: 0x3fffa7018a23)
_ZN4llvm8FastISel17selectInstructionEPKNS_11InstructionE at /home/valentin/julia/usr/bin/../lib/libLLVM-3.8.so (unknown line)
_ZN4llvm16SelectionDAGISel20SelectAllBasicBlocksERKNS_8FunctionE at /home/valentin/julia/usr/bin/../lib/libLLVM-3.8.so (unknown line)
_ZN4llvm16SelectionDAGISel20runOnMachineFunctionERNS_15MachineFunctionE at /home/valentin/julia/usr/bin/../lib/libLLVM-3.8.so (unknown line)
unknown function (ip: 0x3fffa6fbba43)
_ZN4llvm19MachineFunctionPass13runOnFunctionERNS_8FunctionE at /home/valentin/julia/usr/bin/../lib/libLLVM-3.8.so (unknown line)
_ZN4llvm13FPPassManager13runOnFunctionERNS_8FunctionE at /home/valentin/julia/usr/bin/../lib/libLLVM-3.8.so (unknown line)
_ZN4llvm6legacy15PassManagerImpl3runERNS_6ModuleE at /home/valentin/julia/usr/bin/../lib/libLLVM-3.8.so (unknown line)
_ZN4llvm6legacy11PassManager3runERNS_6ModuleE at /home/valentin/julia/usr/bin/../lib/libLLVM-3.8.so (unknown line)
jl_dump_native at /home/valentin/julia/src/jitlayers.cpp:1259
Allocations: 44468446 (Pool: 44467025; Big: 1421); GC: 89

/bin/sh: line 1: 124080 Aborted                 (core dumped) /home/valentin/julia/usr/bin/julia-debug -O0 -C native --output-o /home/valentin/julia/usr/lib/julia/sys-debug.o --startup-file=no --sysimage /home/valentin/julia/usr/lib/julia/inference.ji sysimg.jl
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
make[1]: *** [/home/valentin/julia/usr/lib/julia/sys-debug.o] Error 1
make: *** [julia-sysimg-debug] Error 2
[valentin@ibm-power2 julia]$ 
[valentin@ibm-power2 julia]$ make testall
cp /home/valentin/julia/usr/lib/julia/sys.so /home/valentin/julia/local.so && /home/valentin/julia/usr/bin/julia -J /home/valentin/julia/local.so -e 'true' && rm /home/valentin/julia/local.so
```

cc @ViralBShah

**Comment by vchuravy** [comment]
This seems to be fixed on a build with LLVM 3.9

```julia
[valentin@ibm-power2 julia]$ usr/bin/julia-debug 
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.0-dev.676 (2016-09-16 16:11 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit 628d0e4* (0 days old master)
|__/                   |  powerpc64le-unknown-linux-gnu

julia> versioninfo()
Julia Version 0.6.0-dev.676
Commit 628d0e4* (2016-09-16 16:11 UTC)
DEBUG build
Platform Info:
  System: Linux (powerpc64le-unknown-linux-gnu)
  CPU: unknown
  WORD_SIZE: 64
  BLAS: libopenblas (NO_AFFINITY POWER8)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.0 (ORCJIT, pwr8)
```

**Comment by vchuravy** [comment]
This would also be fixed by upgrading to LLVM 3.9 #18557

**Comment by vchuravy** [comment]
fixed by #18557
